frontend/app/src/view/CreateRoomModal.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useRef, useState } from 'react'
+import {
+  Box,
+  Input,
+  Button,
+  FormControlLabel,
+  RadioGroup,
+  Radio,
+} from '@mui/material'
+import Typography from '@mui/material/Typography'
+import Modal from '@mui/material/Modal'
+import { Socket } from 'socket.io-client'
+
+export const SOCKET_EVENT = {
+  JOIN_ROOM: 'JOIN',
+  SEND_MESSAGE: 'SEND',
+  RECEIVE_MESSAGE: 'RECEIVE',
+  NOICE: 'NOTICE',
+  CREATE: 'CREATE',
+}
+
+type Message = {
+  senderUid: number
+  msgContent: string
+  roomId: string
+}
+
+const style = {
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: 400,
+  bgcolor: 'background.paper',
+  border: '2px solid #000',
+  boxShadow: 24,
+  p: 4,
+}
+
+export const RoomOptionSecond = (prop: {
+  setPassword: (value: string) => void
+  roomType: string
+}) => {
+  const [pwd, setPwd] = useState('false')
+  if (prop.roomType === 'Public') {
+    return (
+      <RadioGroup
+        row
+        aria-labelledby="demo-row-radio-buttons-group-label"
+        name="row-radio-buttons-group"
+        value={pwd}
+        onChange={(e) => setPwd(e.target.value)}
+      >
+        <FormControlLabel
+          value="false"
+          control={<Radio />}
+          label="암호미설정"
+        />
+        <FormControlLabel value="true" control={<Radio />} label="암호설정" />
+        {pwd === 'true' ? (
+          <Input
+            onChange={(e) => prop.setPassword(e.target.value)}
+            placeholder="암호를 입력해주세요"
+          ></Input>
+        ) : (
+          <></>
+        )}
+      </RadioGroup>
+    )
+  } else {
+    return (
+      <Typography
+        id="modal-modal-title"
+        variant="caption"
+        component="h2"
+        height={42}
+        align="justify"
+      >
+        참여 가능한 채팅 리스트에 표시되지 않습니다.
+      </Typography>
+    )
+  }
+}
+
+export default function BasicModal(prop: {
+  modal: boolean
+  setModal: (value: boolean) => void
+  socket: any
+}) {
```

Comment:
```suggestion
export const BasicModal = (prop: {
  modal: boolean
  setModal: (value: boolean) => void
  socket: any
}) => {
```
되도록이면 default보다는 모듈 (`index.ts`)를 사용해서 import 하는 편이 좋을 것 같습니다